src/GeneratorSelector.php

```diff
@@ -46,7 +46,9 @@ public function getGenerator( $format ) {
 		} elseif ( $format === 'atom' ) {
 			return new AtomGenerator();
 		} else {
-			throw new WSExportInvalidArgumentException( "The file format '$format' is unknown." );
+			$msgFormat = '"%s" is not a valid format. Valid formats are: %s';
+			$msg = sprintf( $msgFormat, $format, '"' . implode( '", "', array_keys( self::$formats ) ) . '"' );
+			throw new NotFoundHttpException( $msg );
```

Comment:
Why did you decide to use `NotFoundHttpException` and not something like InvalidArgumentException? Shouldn't an error caused by the client result in a 400?
In any case, if `WSExportInvalidArgumentException` is not in use anymore it should probably be removed

Reply:
Yeah, I wondered about that, and really this change is not the most beautiful. I used a NotFoundHttpException because it's caught by the error-display system that we're already using for "revision not found" etc. Really, it'd probably be better to not have any HTTP stuff inside these classes, and keep all that in the controller; does that sound better?

Reply:
I would say so, yes. Also, `WSExportInvalidArgumentException` should be removed if moving away from it; I could not find any other uses besides this class

Reply:
Done. See what you think. It's slightly odd now, perhaps, in that it's duplicating the message — but I figure that that'll be sorted out when we add i18n to it all.